drivers/sensor/tdk/icp101xx/Devices/Drivers/ICP101xx/Icp101xx.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2017 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/** @defgroup DriverIcp101xx Icp101xx driver
+ *  @brief    Low-level driver for Icp101xx devices
+ *  @ingroup  Drivers
+ *  @{
+ */
+
+#ifndef _INV_ICP101XX_H_
+#define _INV_ICP101XX_H_
+
+#include "../../../EmbUtils/InvExport.h"
+#include "../../../EmbUtils/InvBool.h"
+#include "../../../EmbUtils/InvError.h"
+
+#include "Icp101xxSerif.h"
+
+#include <stdint.h>
+#include <assert.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define ICP101XX_I2C_ADDR (0x63)
```

Comment:
I'm not sure I agree in this case that the vendor code is that much of a benefit. The regmap and functionality here to manipulate the device could easily be made to better fit zephyr's sensor drivers already in the tree without a lot of effort.

Please take a look at drivers like the bmi160 and bme680 which are much closer to how I'd expect to see a sensor driver for Zephyr look

Reply:
The goal of his PR is to make sure Zephyr driver for ICP101xx sensors is using the official driver from TDK.
This code is part of this official TDK driver and cannot be modified for Zephyr.